sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -160,17 +166,22 @@ case class TakeOrdered(limit: Int, sortOrder: Seq[SortOrder], child: SparkPlan)
 
   private val ord: RowOrdering = new RowOrdering(sortOrder, child.output)
 
-  private def collectData(): Array[InternalRow] =
-    child.execute().map(_.copy()).takeOrdered(limit)(ord)
+  @transient private val projection = projectList.map(newProjection(_, child.output))
```

Comment:
How did you see errors here?  I removed `@transient` and `sbt sql/test` still works for me.